runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1449,9 +1449,24 @@ TR_VectorAPIExpansion::isVectorizedOrScalarizedNode(TR::Node *node, TR::DataType
    int32_t refId = -1;
 
    if (node->getOpCodeValue() == TR::aload ||
-       node->getOpCodeValue() == TR::astore)
-      {
-      refId = node->getSymbolReference()->getReferenceNumber();
+       node->getOpCodeValue() == TR::astore ||
+       (node->getOpCode().isVectorOpCode() &&
+        (node->getOpCode().getVectorOperation() == TR::vload ||
+         node->getOpCode().getVectorOperation() == TR::vstore)
```

Comment:
`vload` or `vstore` is what I would (perhaps naively) expect to see for any vectorized load or store, so I suppose I don't understand why `aload`, `astore`, and calls are accepted. Is this detecting nodes that have already been selected by the current pass for vectorization/scalarization but that may or may not have been transformed yet?

Reply:
The method checks it was decided by the analysis to vectorize the node. In some rare cases (eg. commoning) node could be already transformed. This fix is for that case.